src/dotnet/commands/dotnet-build/ProjectGraphCollector.cs

```diff
@@ -44,7 +44,7 @@ private ProjectGraphNode TraverseProject(ProjectDescription project, IDictionary
                     LibraryDescription libraryDescription;
                     if (lookup.TryGetValue(dependency.Name, out libraryDescription))
                     {
-                        if (libraryDescription.Identity.Type.Equals(LibraryType.Project))
+                        if (libraryDescription.Resolved && libraryDescription.Identity.Type.Equals(LibraryType.Project))
```

Comment:
Shouldn't it be an error if the project is not resolved?

Right now, if a project is not resolved the else branch is taken which implicitly assumes that it receives a non ProjectDescription. This might lead to a wrong graph.